File: src/main/java/com/dg/watcher/history/graph/HistoryGraphUrlGenerator.java
package com.dg.watcher.history.graph;

import hudson.model.AbstractProject;
import org.jfree.chart.urls.CategoryURLGenerator;
import org.jfree.data.category.CategoryDataset;


public class HistoryGraphUrlGenerator implements CategoryURLGenerator {
    private AbstractProject<?, ?> project;


    public HistoryGraphUrlGenerator(AbstractProject<?, ?> project) {
        this.project = project;
    }

    @Override
    public String generateURL(CategoryDataset categoryDataset, int series, int itemIndex) {
        return project.getAbsoluteUrl() + retrieveBuildNumber(categoryDataset, itemIndex);
    }

    private String retrieveBuildNumber(CategoryDataset categoryDataset, int itemIndex) {
        return ((String) categoryDataset.getColumnKey(itemIndex)).replaceFirst("#", "");
    }
}

File: src/main/java/com/dg/watcher/history/graph/HistoryGraphUrlGenerator.kt
package com.dg.watcher.history.graph

import com.dg.watcher.base.Project
import org.jfree.chart.urls.CategoryURLGenerator
import org.jfree.data.category.CategoryDataset


class HistoryGraphUrlGenerator(private val project: Project) : CategoryURLGenerator {
    override fun generateURL(categoryDataSet: CategoryDataset, series: Int, itemIndex: Int) =
            project.getAbsoluteUrl() + getBuildNumber(categoryDataSet, itemIndex)

    private fun getBuildNumber(categoryDataSet: CategoryDataset, itemIndex: Int) =
            (categoryDataSet.getColumnKey(itemIndex) as String).replaceFirst("#", "")
}